packages/wdio-jasmine-framework/src/index.ts

```diff
@@ -201,6 +201,7 @@ class JasmineAdapter {
             self._lastTest = this.result
             // @ts-ignore overwrite existing type
             self._lastTest.start = new Date().getTime()
+            globalThis._wdioDynamicJasmineResult = this.result
```

Comment:
Then maybe do this her:
```suggestion
            if (this.result.passedExpectations.length === 0 && this.result.failedExpectations.length > 0) {
                globalThis._wdioDynamicJasmineResult = this.result.failedExpectations[0]
            }
```

Reply:
Here in `init()` we simply get a reference to a variable with the test results. But here `result.failedExpectations/passedExpectations` are empty arrays, so the condition will never be satisfied and the assignment will not take place.

As far as I can see, the results end up in the variable `result` dynamically only after this step: 
```typescript
// testFnWrapper.ts

 result = await executeAsync.call(this, specFn, retries, specFnArgs, timeout)
``` 

Therefore, it seems to me that it would be more appropriate to leave only the variable assignment here, as it was before. And then already do the conditions and checks in `testFnWrapper.ts`